src/gcp/cloudscheduler.ts

```diff
@@ -12,38 +14,113 @@ export interface Schedule {
     uri: string;
     httpMethod: string;
   };
+  retryConfig?: {
+    retryCount?: number;
+    maxRetryDuration?: string;
+    minBackoffDuration?: string;
+    maxBackoffDuration?: string;
+    maxDoublings?: number;
+  };
 }
 
-export function createJob(schedule: Schedule): Promise<void> {
+/*
+* Creates a cloudScheduler job.
+* If another job with that name already exists, this will return a 409.
+* @param job: The job to create
+*/
+export function createJob(job: Job): Promise<void> {
   // the replace below removes the portion of the schedule name after the last /
   // ie: projects/my-proj/locations/us-central1/jobs/firebase-schedule-func-us-east1 would become
   // projects/my-proj/locations/us-central1/jobs
-  const strippedName = schedule.name.replace(/\/[^\/]+$/, "");
+  const strippedName = job.name.replace(/\/[^\/]+$/, "");
   return api.request("POST", `/${VERSION}/${strippedName}`, {
     auth: true,
     origin: api.cloudschedulerOrigin,
-    data: Object.assign({ timeZone: "America/Los_Angeles" }, schedule),
+    data: Object.assign({ timeZone: DEFAULT_TIME_ZONE }, job),
   });
 }
 
+/*
+* Deletes a cloudScheduler job with the given name.
+* Returns a 404 if no job with that name exists.
+* @param name: The name of the job to delete
+*/
 export function deleteJob(name: string): Promise<void> {
   return api.request("DELETE", `/${VERSION}/${name}`, {
     auth: true,
     origin: api.cloudschedulerOrigin,
   });
 }
 
-export async function createOrReplaceJob(schedule: Schedule): Promise<void> {
-  const jobName = `${schedule.name.split("/")[5]}`;
+/*
+* Gets a cloudScheduler job with the given name.
+* If no job with that name exists, this will return a 404.
+* @param name: The name of the job to get
+*/
+export function getJob(name: string): Promise<any> {
+  return api.request("GET", `/${VERSION}/${name}`, {
+    auth: true,
+    origin: api.cloudschedulerOrigin,
+  });
+}
+
+/*
+* Updates a cloudScheduler job.
+* Returns a 404 if no job with that name exists.
+* @param job: A job to update. Note that name cannot be updated.
+*/
+export function updateJob(job: Job): Promise<any> {
```

Comment:
Added this after chatting with michael - he originally used delete + create instead since update was unable to change one of the fields he wanted to change (maybe name?). However, I tested manually tested changing the 3 fields that we care about here (schedule, timeZone, and retryConfig), and it worked fine, so I switched it over to a patch call.

Reply:
This comment should live in the code, something along the lines of "the tested fields that can be updated are [...]. Use caution when adding other fields to be updated."